src/services/sync/strategies/TSP/TSPBaseSyncer.js

```diff
@@ -160,7 +160,6 @@ class TSPBaseSyncer extends Syncer {
 		return this.app.service('schools').create({
 			name,
 			systems: [this.tspSystemId],
-			currentYear: await this.getCurrentYear(),
```

Comment:
Damit verschwindet allerdings der throw sollte kein year in der DB vorhanden sein. Ob das sinnvoll ist, keine Ahnung, ob das nicht Teil der Facade / create hooks sein muss sei auch dahin gestellt, wollte aber darauf hinweisen.

Reply:
Ich bin auch nicht sicher, ob das extra geprüft werden muss, aber ich hab jetzt einfach im Hook einen solchen Check eingebaut. Allerdings mit `years.length`. Was soll `years.total` überhaupt sein?

Reply:
mongoose (die mongoDB?, mongoose-featherJS?) liefert immer beim find ein paginated object zurück. Daher { data, total, skip?, limit? } . Solange das nicht irgendwo umgemapped wird.

Reply:
Ich finde dazu irgendwie nichts. Aber bei einer query mit lean-Option wie im Hook muss es auf jeden Fall length sein.